src/samples/plugins/TokenSessionKeyPlugin.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.19;
+
+import {
+    ManifestFunction,
+    ManifestAssociatedFunctionType,
+    ManifestAssociatedFunction,
+    PluginManifest,
+    PluginMetadata,
+    SelectorPermission,
+    ManifestExternalCallPermission
+} from "../../interfaces/IPlugin.sol";
+import {BasePlugin} from "../../plugins/BasePlugin.sol";
+import {ModularSessionKeyPlugin} from "./ModularSessionKeyPlugin.sol";
+import {ITokenSessionKeyPlugin} from "./interfaces/ITokenSessionKeyPlugin.sol";
+import {IModularSessionKeyPlugin} from "./interfaces/ISessionKeyPlugin.sol";
+import {IPluginExecutor} from "../../interfaces/IPluginExecutor.sol";
+
+/// @title Token Session Key Plugin
+/// @author Decipher ERC-6900 Team
+/// @notice This plugin acts as a 'child plugin' for ModularSessionKeyPlugin.
+/// It implements the logic for session keys that are allowed to call ERC20
+/// transferFrom function. It allows for session key owners to access MSCA
+/// with `transferFromSessionKey` function, which calls `executeFromPluginExternal`
+/// function in PluginExecutor contract.
+/// The target ERC20 contract and the selector for transferFrom function are hardcoded
+/// in this plugin, since the pluginManifest function requires the information of
+/// permitted external calls not to be changed in the future. For other child session
+/// key plugins, there can be a set of permitted external calls according to the
+/// specific needs.
+contract TokenSessionKeyPlugin is BasePlugin, ITokenSessionKeyPlugin {
+    string public constant NAME = "Token Session Key Plugin";
+    string public constant VERSION = "1.0.0";
+    string public constant AUTHOR = "Decipher ERC-6900 Team";
+
+    // Mock address of target ERC20 contract
+    address public constant TARGET_ERC20_CONTRACT = 0xdeaDDeADDEaDdeaDdEAddEADDEAdDeadDEADDEaD;
+    bytes4 public constant TRANSFERFROM_SELECTOR =
+        bytes4(keccak256(bytes("transferFrom(address,address,uint256)")));
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Execution functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc ITokenSessionKeyPlugin
+    function transferFromSessionKey(address target, address from, address to, uint256 amount)
```

Comment:
`target` seems redundant since only `TARGET_ERC20_CONTRACT` is valid, but if this is to showcase external call permissions, this is fine.

Reply:
Yep that was to show the wrong call to unregistered contract fails in the test code.